### PR TITLE
Extra observables

### PR DESCRIPTION
**Pull request by rvanheest**
I have been using RxJavaFX for a couple of months now and often came across limitations of this library. To solve these limitations, I added a number of wrappers that must be in this library to support the collaboration between RxJava and JavaFX more optimally.

First of all, besides many properties/`ObservableValues`, JavaFX has a number of observable collections: `ObservableList`, `ObservableMap` and `ObservableSet` that can be turned into Rx Observables. Every time a collection fires a change event, a new snapshot of that collection is put into the Rx Observable.

Secondly, the RxJavaFX library was already capable of fetching all events of a certain type fired from a specific `Node`. However, this was not possible for something like a `Stage` or other type of `Window`. Therefore I added a wrapper that supports fetching a certain type of events from a `Window`.

Let me know what you think about this and what improvements there could be on the new code.

### CODE REVIEW

**Comment by thomasnield** [comment]
I know this is an old PR, but I left it open to give it some thought after taking ownership of this library. Here is what I think.

RxJavaFX has simple factories to turn _any_ `ObservableValue`, `Property`, or `Node` event into an RxJava `Observable`. Trying to build API's around specific controls or windows is basically doing what the client can do already. Therefore it might be overkill to do this, and we should not undermine the simplicity of this library without significant functionality gain. 

Then there are the `ObservableCollections` you listed. While it may seem logical at first to turn changes on an `ObservableList` into RxJava `Observable` emissions, there is a bigger question that needs to be asked: what is triggering the changes to the `ObservableList`?  If it is backing a `TableView`, why are changes to the`ObservableList` the source of `Observable` emissions? In a true reactive application, should not the thing changing the `ObservableList` be the source `Observable`? The `TableView` and its `ObservableList` should be subscribing to an `Observable`, not being one. Anything else interested in these events would subscribe to that source as well rather than going to the `ObservableList` acting as a middleman. 

Now there is the [ReactFX](https://github.com/TomasMikula/ReactFX) library where functionalities like this might make sense. ReactFX is entirely JavaFX-focused and all of its reactive streams are built on UI events. I would check that library out if JavaFX is your sole focus. If JavaFX is simply a means-to-an-end layer on a complex application, RxJavaFX might be a better choice. 

In a nutshell, this library provides a simple means to make JavaFX a client to RxJava, not the other way around. 

Unless anyone has anything to add, I propose closing this.

**Comment by thomasnield** [comment]
I had a change of heart and I might explore this since some events (like selected rows in a `TableView`) only manifest in the form of `ObservableList`. I think the design should be done differently though. I think individual items should be emitted and categorized with the type of change. I'll explore this later.
